File: misc/hackerrank/cs5/fibo.py
import math
t = raw_input()
t = int(t)

def isperfect(p):
	x = math.sqrt(p)
	return (x*x == p)

def isfibo(n):
	return isperfect((5*n*n) - 4) or isperfect((5*n*n)+4)

while t:
	n = raw_input()
	n = int(n)
	if isfibo(n):
		print "IsFibo"
	else:
		print "IsNotFibo"
	t = t - 1

File: misc/hackerrank/cs5/fibo.cpp
#include <iostream>
#include <stdio.h>
#include <cmath>
using namespace std;
 
bool isPerfectSquare(long long int x)
{
    cout<<x<<endl;
    long long int s = sqrt(x);
    return (s*s == x);
}
 
bool isFibonacci(long int n)
{
    return isPerfectSquare(5*n*n + 4) ||
           isPerfectSquare(5*n*n - 4);
}
 
int main(){
  int t;
  cin >> t;
  while(t--){
    long int n;
    cin >> n;
    if(isFibonacci(n))
      cout<<"IsFibo"<<endl;
    else
      cout<<"IsNotFibo"<<endl;
  }
  return 0;
}